modules/react/loading-dots/lib/LoadingDots.tsx

```diff
@@ -1,73 +1,83 @@
 import * as React from 'react';
-import {keyframes} from '@emotion/react';
-import {canvas, borderRadius} from '@workday/canvas-kit-react/tokens';
-import {styled} from '@workday/canvas-kit-react/common';
+import {keyframes} from '@emotion/css';
+import {base, system} from '@workday/canvas-tokens-web';
+import {createComponent} from '@workday/canvas-kit-react/common';
+import {createStyles, handleCsProp, CSProps} from '@workday/canvas-kit-styling';
+
+/**
+ * Duration of the sparkle dots loading (in ms).
+ */
+const ANIMATION_DURATION_MS = '1400';
 
 /**
  * Keyframe for the dots loading animation.
  */
-const keyframesLoading = keyframes`
-  0%,
-  80%,
-  100% {
-    transform: scale(0);
-}
-  40% {
-    transform: scale(1);
-}`;
+const keyframesLoading = keyframes({
+  '0%, 80%, 100%': {
+    transform: 'scale(0)',
+  },
+  '40%': {
+    transform: 'scale(1)',
+  },
+});
 
-type LoadingDotProps = {
-  /**
-   * The animation delay of the LoadingDots (in ms).
-   */
-  animationDelay: number;
-};
+const loadingDotStyles = createStyles({
+  backgroundColor: base.soap400,
+  width: system.space.x4,
+  height: system.space.x4,
+  fontSize: system.space.zero,
+  borderRadius: system.shape.round,
+  transform: 'scale(0)',
+  display: 'inline-block',
+  marginRight: system.space.x2,
+  animationName: keyframesLoading,
+  animationDuration: `${ANIMATION_DURATION_MS}ms`,
+  animationIterationCount: 'infinite',
+  animationTimingFunction: 'ease-in-out',
+  animationFillMode: 'both',
+  '&:last-child': {
+    marginRight: 0,
+  },
+  '&:nth-child(1)': {
+    animationDelay: '0ms',
+  },
+  '&:nth-child(2)': {
+    animationDelay: `calc(${ANIMATION_DURATION_MS}ms * (4/35))`, // 160
+  },
+  '&:nth-child(3)': {
+    animationDelay: `calc(${ANIMATION_DURATION_MS}ms * (8/35))`, // 320
+  },
+});
 
 /**
  * The actual loading dot div.
  */
-const LoadingAnimationDot = styled('div')<LoadingDotProps>(
-  {
-    backgroundColor: canvas.colors.soap400,
-    width: canvas.space.s,
-    height: canvas.space.s,
-    fontSize: '0px',
-    borderRadius: borderRadius.circle,
-    transform: 'scale(0)',
-    display: 'inline-block',
-    marginRight: canvas.space.xxs,
-    animationName: keyframesLoading,
-    animationDuration: '1400ms',
-    animationIterationCount: 'infinite',
-    animationTimingFunction: 'ease-in-out',
-    animationFillMode: 'both',
-    '&:last-child': {
-      marginRight: 0,
-    },
+const LoadingAnimationDot = createComponent('div')({
```

Comment:
const LoadingAnimationDot = () =>  <div className={`${loadingDotStyles}`}></div>

This component is not going to be used out of the LoadingDots, so was also thinking we could make it a simple div. 

Any recommendation on the approach that would fit best?

Reply:
It can be a simple `div`. You can simplify the className:

```jsx
const LoadingAnimationDot = () => <div className={loadingDotStyles} />;
```

Reply:
There isn't much abstraction here. The `<div>` could be inlined in `LoadingDots`, but it doesn't matter much. There isn't any advantage to using the `createComponent` abstraction and it isn't exported, so a simple component is fine. I'm also fine with inlining the `div` elements.